package/src/Guard.tsx

```diff
@@ -24,13 +24,22 @@ interface GuardsResolve {
   redirect: RouteRedirect;
 }
 
-const Guard: React.FunctionComponent<GuardProps> = ({ children, component, meta, render }) => {
+const Guard: React.FunctionComponent<GuardProps> = ({
+  children,
+  component,
+  meta,
+  render,
+  pathChanged,
+  name,
+}) => {
   const routeProps = useContext(RouterContext);
   const routePrevProps = usePrevious(routeProps);
-  const hasPathChanged = useMemo(
-    () => routeProps.location.pathname !== routePrevProps.location.pathname,
-    [routePrevProps, routeProps],
-  );
+  const hasPathChanged = useMemo(() => {
+    if (pathChanged && typeof pathChanged === 'function') {
```

Comment:
It'd be more clear if we named this variable `checkPathChanged` so you know from a glance it should be a function